processor/error/event.go

```diff
@@ -123,40 +124,68 @@ func (e *Event) addGroupingKey() {
 	e.add("grouping_key", e.calcGroupingKey())
 }
 
-func (e *Event) calcGroupingKey() string {
-	hash := md5.New()
+type groupingKey struct {
+	hash  hash.Hash
+	empty bool
+}
 
-	add := func(s *string) bool {
-		if s != nil {
-			io.WriteString(hash, *s)
-		}
-		return s != nil
+func newGroupingKey() *groupingKey {
+	return &groupingKey{
+		hash:  md5.New(),
+		empty: true,
+	}
+}
+
+func (k *groupingKey) add(s *string) bool {
```

Comment:
these methods will panic if k is `nil` but since it's private and used in exactly one place I skipped guards.